src/fips_check.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * This program is released under the Common Public License V1.0
+ *
+ * You should have received a copy of Common Public License V1.0 along with
+ * with this program.
+ *
+ * Copyright IBM Corp. 2020
+ */
+
+
+#ifdef ICA_FIPS
+
+#include <errno.h>
+#include <openssl/crypto.h>
+#include <openssl/hmac.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <syslog.h>
+#include <dlfcn.h>
+
+
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
+#define HMAC_PREFIX "."
+#ifndef HMAC_SUFFIX
+#define HMAC_SUFFIX ".hmac"
+#endif
+#define READ_BUFFER_LENGTH 16384
+
+static const char hmackey[] = "00000000000000000000000000000000";
```

Comment:
I'd like to see a comment here how this hmackey is actually filled with the real value.

Reply:
Done.